src/Collections/Sequences/Seq.dfy

```diff
@@ -559,6 +559,7 @@ module Seq {
     if |s| == 0 {
     } else {
       LemmaFlattenLengthLeMul(s[..|s|-1], j);
+      assert |FlattenReverse(s[..|s|-1])| <= (|s|-1) * j;
```

Comment:
This assert just repeats the ensure, why is it needed?

Reply:
It's not exactly restating the postcondition: it's stating a more specific instance of it that also implies an equality. I think this general pattern, asserting an instantiation of a more general postcondition after a lemma call, is one of the steps we might need to start using more often to get proofs that verify more predictably.